pytorch_lightning/core/sampler.py

```diff
@@ -0,0 +1,61 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from torch.utils.data import IterableDataset
+from torch.utils.data.dataloader import DataLoader
+from torch.utils.data.sampler import BatchSampler
+
+
```

Comment:
Instead of using a special sampler (which in my opinion is not necessary here), can't we just auto_id by the batch_idx and the number of samples per batch?

If the user wants it to be consistent he should either provide an is himself or just sample sequentially. But IMO a separate sampler is a bit overkill and we shouldn't alter the users code and choices to much.

Reply:
This won't work in distributed setting.

Reply:
But the current approach works when running using a DistributedSampler.

Reply:
Great idea ! Done !